okhttp/src/main/java/okhttp3/internal/platform/android/Android10CertificateChainCleaner.kt

```diff
@@ -52,4 +53,14 @@ internal class Android10CertificateChainCleaner(
           other.trustManager === this.trustManager
 
   override fun hashCode(): Int = System.identityHashCode(trustManager)
+
+  companion object {
+    fun buildIfSupported(trustManager: X509TrustManager): Android10CertificateChainCleaner? {
+      return try {
+        Android10CertificateChainCleaner(trustManager)
+      } catch (iae: IllegalArgumentException) {
+        null
```

Comment:
add a comment pointing at why this is a reasonable thing to do here?

Reply:
also – could potentially limit try/catch to creating the `X509TrustManagerExtensions` instance, and then pass _that_ to the `Android10CertificateChainCleaner` constructor